sdk/nodejs/appservice/zMixins.ts

```diff
@@ -312,15 +374,45 @@ function redirectConsoleOutput<C extends Context<R>, E, R extends Result>(callba
     };
 }
 
-
 /**
-  * A CallbackFunctionApp is a special type of azure.appservice.FunctionApp that can be created out
-  * of an actual JavaScript function instance.  The function instance will be analyzed and packaged
-  * up (including dependencies) into a form that can be used by AWS Lambda.  See
-  * https://github.com/pulumi/docs/blob/master/reference/serializing-functions.md for additional
-  * details on this process.
+ * Interface to define individual Azure Functions. A function is not deployable by itself: instead, it must be passed
+ * to MultiFunctionApp to be deployed as a part of the Azure Function App.
  */
-export class CallbackFunctionApp<C extends Context<R>, E, R extends Result> extends FunctionApp {
+export interface Function {
+    /**
+     * Azure Function name.
+     */
+    name: string;
+
+    /**
+     * An array of function binding definitions.
+     */
+    bindings: pulumi.Input<BindingDefinition[]>;
+
+    /**
+     * Function callback.
+     */
+    callback: CallbackArgs<Context<any>, any, any>;
+
+    /**
+     * Application settings required by the function.
+     */
+    appSettings?: pulumi.Input<{ [key: string]: any }>;
+}
+
+export interface MultiFunctionAppArgs extends FunctionAppArgsBase {
+    /**
+     * The functions to deploy as parts of this application. Either one of [functions] or [archive] must be provided.
+     */
+    functions?: Function[];
+
+    /**
+     * The deployment package to deploy as-is. Either one of [functions] or [archive] must be provided.
+     */
+    archive?: pulumi.Input<pulumi.asset.Archive>;
+};
+
+export class MultiFunctionApp extends FunctionApp {

```

Comment:
ok... so here's where i'm landing with this.  

First: because `MultiFunctionApp` takes in an array of `Function`s, and because each `Function` wraps a `Callback`, I think this really is a `MultiCallbackFunctionApp`.  (Or it's just `CallbackFunctionApp`, and we make it so that `CallbackFunctionApp` can take one or many callbacks).

Second: II think 'archive' muddies the waters here and makes it less clear what hte purpose of this all is.  I think i'd be fine with ArchiveFunctionApp being something else entirely.  That way there are clear, separated purposes for all these types.  the `CallbackFunctionApp` (and *possibly* `MultiCallbackFunctionApp` if we have that) is for wrapping up 1-many callbacks into a FunctionApp.  `ArchiveFunctionApp` is for the archive case.



Now, i'd love if we just ended up with CallbackFunctionApp allowing 1-many functions (with constructor overloads for the existing constructor we support).  If that's not possible, i'm ok having that exist for the '1-case', and MultiCallbackFunctionApp exist for the '2-N case'.

Thoughts?

Reply:
How do we unify a generic `CallbackFunctionApp<C, E, R>` with the non-generic `CallbackFunctionApp` with multiple functions?

Reply:
@mikhailshilkov Good point.  We likely can't.  :)  Based on that, it seems ok to have the split there.

Reply:
Note: i think splitting out 'archive' would be fine.